backend-plugin/src/main/java/com/redhat/jenkins/nodesharingbackend/Pool.java

```diff
@@ -107,7 +113,26 @@
         return property;
     }
 
-    public @CheckForNull StandardUsernamePasswordCredentials getCredential() {
+    public @CheckForNull UsernamePasswordCredentials getExecutorCredential(ExecutorJenkins executor) {
+        if(executor.getCredentialId() != null && !executor.getCredentialId().isEmpty()) {
+            UsernamePasswordCredentials cred = CredentialsMatchers.firstOrNull(
+                    CredentialsProvider.lookupCredentials(UsernamePasswordCredentials.class, Jenkins.getInstance(),
+                            ACL.SYSTEM, Collections.<DomainRequirement>emptyList()),
+                    CredentialsMatchers.withId(executor.getCredentialId())
+            );
+
+            LOGGER.info("credential id: " + executor.getCredentialId());
```

Comment:
This probably does not deserve to be logged at `info` level. Also, please add the name of executor so the logs can be tied to resource identity.

Reply:
this is leftover from my work which I forgot to remove. But it may be useful to change level to finest and give more details so in case of problems user can see this message.

Reply:
Should we preserve a backward compatibility? If yes, we should leave ```getCredential()``` untouched and introduce a new ```getExecutorCredential()``` method.

Reply:
I found that getCredential() was used only in Pool.java and changed name to better describe what type of credential is returned. Do you think that there are other plugins using getCredential() ?

Reply:
I understand you however I was taught by other if something is a part of public API it's out of my control who and when uses that part. But if @olivergondza is fine with that name change, lets go...